doc/radosgw/d3n_datacache.rst

```diff
@@ -0,0 +1,107 @@
+
+| D3N improves the performance of big-data jobs running in analysis clusters by speeding up recurring reads from the data lake.
+| The Rados Gateways act as cache servers for the back-end object store (OSDs), storing data locally for reuse.
+
+Architecture
+============
+
+D3N improves the performance of big-data jobs by speeding up repeatedly accessed dataset reads from the data lake.
+Cache servers are located in the datacenter on the access side of potential network and storage bottlenecks,
+D3Ns two layer logical caches forms a traditional caching hierarchy :sup:`*`
+where caches nearer the client have the lowest access latency and overhead,
+while caches in higher levels in the hierarchy are slower (requiring multiple hops to access),
+The layer 1 cache server nearest to the client handles object requests by breaking them into blocks,
+returning any blocks which are cached locally, and forwarding missed requests to the block home location
+(as determined by consistent hashing) in the next layer.
+Cache misses are forwarded to successive logical caching layers until a miss at the top layer is resolved
+by a request to the data lake (Rados)
+
+:sup:`*` currently only layer 1 cache has been upstreamed.
+
+See `MOC D3N (Datacenter-scale Data Delivery Network)`_ and `Red Hat Research D3N Cache for Data Centers`_.
+
+Implementation
+==============
+
+- D3N cache supports both the `S3` and `Swift` object storage interfaces.
+- D3N will cache tail objects because they are immutable (by default it is parts of objects that are larger than 4MB).
+
+Requirements
+------------
+
+- A SSD(/dev/nvme,/dev/pmem,/dev/shm) storage device, formatted (filesystems other than XFS were not tested) and mounted,
+  it will be used as the cache backing store.
+  (depending on device performance, multiple RGWs may share a single device but each requires a discrete directory on the device)
+
+Limitations
+-----------
+
+- D3N will not cache objects compressed by `Rados Gateway Compression`_ (OSD level compression is supported).
+- D3N will not cache objects encrypted by `Rados Gateway Encryption`_.
+- D3N will be disabled if the `rgw_max_chunk_size` config variable value differs from the `rgw_obj_stripe_size` config variable value.
+
+
+D3N Environment Setup
+=====================
+
+Running
+-------
+
+D3N minimal configuration startup example using `vstart.sh`::
+
+    env MON=1 OSD=1 MDS=0 MGR=1 RGW=1 ../src/vstart.sh -n  \
+    -o rgw_d3n_l1_local_datacache_enabled=true  \
+    -o rgw_d3n_l1_datacache_persistent_path=/mnt/nvme0/rgw_datacache/  \
+    -o rgw_d3n_l1_datacache_size=10737418240
+
+the above example assumes that the cache backing-store solid state device
+is mounted at `/mnt/nvme0` and has `10 GB` of free space available.
+
+    NOTE: Each time the Rados Gateway is restarted the content of the cache directory is purged.
+
+Logs
+----
+- D3N related log lines in `radosgw.*.log` contain the string `d3n` (case insensitive).
```

Comment:
in what way is "d3n" case insensitive in a log line?

Reply:
i.e, case insensitivity can be a property of a filesystem or collation system, but not of a string literal

Reply:
some of the logs have `... D3N ...` and some of the logs `... d3n ...`, will add to the D3N talk list a log cleanup PR